lib/cstruct.ml

```diff
@@ -96,7 +96,7 @@ let create_unsafe len =
   { buffer ; len ; off = 0 }
 
 let check_bounds t len =
-  Bigarray.Array1.dim t.buffer >= len
+  len >= 0 && Bigarray.Array1.dim t.buffer >= len
```

Comment:
@talex5: I think the docstring for `check_bounds` needs to be updated to mention the check that `len` is positive.